ospf6d/ospf6_asbr.c

```diff
@@ -1837,31 +1838,65 @@ static int ospf6_as_external_lsa_show(struct vty *vty, struct ospf6_lsa *lsa)
 		 (CHECK_FLAG(external->bits_metric, OSPF6_ASBR_BIT_T) ? 'T'
 								      : '-'));
 
-	vty_out(vty, "     Bits: %s\n", buf);
-	vty_out(vty, "     Metric: %5lu\n",
-		(unsigned long)OSPF6_ASBR_METRIC(external));
-
-	ospf6_prefix_options_printbuf(external->prefix.prefix_options, buf,
-				      sizeof(buf));
-	vty_out(vty, "     Prefix Options: %s\n", buf);
+	if (use_json) {
+		json_object_string_add(json_obj, "bits", buf);
+		json_object_int_add(json_obj, "metric",
+				    (unsigned long)OSPF6_ASBR_METRIC(external));
+		ospf6_prefix_options_printbuf(external->prefix.prefix_options,
+					      buf, sizeof(buf));
+		json_object_string_add(json_obj, "prefixOptions", buf);
+		json_object_int_add(
+			json_obj, "referenceLsType",
+			ntohs(external->prefix.prefix_refer_lstype));
+		json_object_string_add(json_obj, "prefix",
+				       ospf6_as_external_lsa_get_prefix_str(
+					       lsa, buf, sizeof(buf), 0));
+
+		/* Forwarding-Address */
+		json_object_boolean_add(
+			json_obj, "forwardingAddressPresent",
+			CHECK_FLAG(external->bits_metric, OSPF6_ASBR_BIT_F));
+		if (CHECK_FLAG(external->bits_metric, OSPF6_ASBR_BIT_F))
+			json_object_string_add(
+				json_obj, "forwardingAddress",
+				ospf6_as_external_lsa_get_prefix_str(
+					lsa, buf, sizeof(buf), 1));
+
+		/* Tag */
+		json_object_boolean_add(
+			json_obj, "tagPresent",
+			CHECK_FLAG(external->bits_metric, OSPF6_ASBR_BIT_T));
+		if (CHECK_FLAG(external->bits_metric, OSPF6_ASBR_BIT_T))
+			json_object_int_add(json_obj, "tag",
+					    ospf6_as_external_lsa_get_tag(lsa));
+	} else {
+		vty_out(vty, "     Bits: %s\n", buf);
+		vty_out(vty, "     Metric: %5lu\n",
+			(unsigned long)OSPF6_ASBR_METRIC(external));
 
-	vty_out(vty, "     Referenced LSType: %d\n",
-		ntohs(external->prefix.prefix_refer_lstype));
+		ospf6_prefix_options_printbuf(external->prefix.prefix_options,
+					      buf, sizeof(buf));
+		vty_out(vty, "     Prefix Options: %s\n", buf);
 
-	vty_out(vty, "     Prefix: %s\n",
-		ospf6_as_external_lsa_get_prefix_str(lsa, buf, sizeof(buf), 0));
+		vty_out(vty, "     Referenced LSType: %d\n",
```

Comment:
minor indent issue

Reply:
This is auto indented after running the clang tool